src/transport_ofi.h

```diff
@@ -460,8 +458,12 @@ static inline
 void
 shmem_transport_put_wait(long *completion) {
 
-    if((*completion)-- > 0)
+    shmem_internal_assertp((*completion) >= 0);
```

Comment:
It should be fine to use a regular assertion here.  The ```p``` in ```assertp``` will cause it to persist even when error checking is disabled.